pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -182,3 +182,31 @@ func (c *ClusterController) updateClusterCephVersion(image string, cephVersion c
 		return
 	}
 }
+
+func (c *ClusterController) UpdateCephStorage(client client.Client, name types.NamespacedName) {
```

Comment:
I'd suggest refact https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/controller.go#L483 instead of duplicating code.

Reply:
Agreed, adding to that method sounds good. But we also don't want to query ceph every time there is a status update. We only should query for the device classes at the end of a successful reconcile. So perhaps the `updateStats()` method could have a flag that indicates if device classes should be updated, and it would only be set to true after the successful reconcile.

Reply:
Few clarifications on adding into `updateStatus` method

1. `NewCephCommand` method requires `context` parameter in order to process the query, not quite clear on how we can pass in the `context` from `updateStatus` method
2. AFAIK, the above updateStatus method is only invoked for updating ceph cluster status to `Deleting`

Reply:
After looking at this again, let's not merge the updateStatus() implementations for now. Different parts of the status are updated on different schedules and change different fields, so this will work well for you. @leseb If you're seeing a refactor, how about a separate PR?

Reply:
Sounds good.